awx/ui/client/lib/components/relaunchButton/relaunchButton.component.js

```diff
@@ -5,17 +5,117 @@ const atRelaunch = {
     bindings: {
         state: '<'
     },
-    controller: ['RelaunchJob', 'InitiatePlaybookRun', 'ComponentsStrings', '$scope', atRelaunchCtrl],
+    controller: ['ProcessErrors', 'AdhocRun', 'ComponentsStrings',
+        'ProjectModel', 'InventorySourceModel', 'WorkflowJobModel', 'Alert',
+        'AdHocCommandModel', 'JobModel', 'JobTemplateModel', 'PromptService',
+        'GetBasePath', '$state', '$q', '$scope', atRelaunchCtrl
+    ],
     controllerAs: 'vm'
 };
 
-function atRelaunchCtrl (RelaunchJob, InitiatePlaybookRun, strings, $scope) {
+function atRelaunchCtrl (
+    ProcessErrors, AdhocRun, strings,
+    Project, InventorySource, WorkflowJob, Alert,
+    AdHocCommand, Job, JobTemplate, PromptService,
+    GetBasePath, $state, $q, $scope
+) {
     const vm = this;
     const scope = $scope.$parent;
     const job = _.get(scope, 'job') || _.get(scope, 'completed_job');
```

Comment:
I would include these as bindings instead of utilizing the parent scope within the controller.

<at-relaunch job="job" completed_job="completed_job"></at-relaunch>